src/Language/Source.php

```diff
@@ -28,7 +33,8 @@ public function __construct($body, $name = null)
         );
 
         $this->body = $body;
-        $this->length = mb_strlen($body, 'UTF-8');
+        $this->buf =  array_map('GraphQL\Utils::ord', preg_split('//u', $body, -1, PREG_SPLIT_NO_EMPTY));
```

Comment:
Hey @vladar, sorry for the silence yesterday, busy weekend! I worked up https://github.com/aelnagger/graphql-php/tree/utf_8_decoder a few days ago, passing all the tests, but haven't cleaned it up or run the benchmarks just yet. There's definitely an issue with garbage collection on larger queries though. Making use of graphQL variables is definitely the right solution.